src/python/Utils/CPMetrics.py

```diff
@@ -0,0 +1,305 @@
+    if isinstance(metrics, str):
+        metrics = json.loads(metrics)
+    # the following keys will be skipped
+    skip = [
+        'cherrypy_app_enabled',
+        'cherrypy_http_server_bind_address',
+        'cherrypy_app_requests',
+        'cherrypy_app_server_version',
+        'cherrypy_http_server_enabled'
+    ]
+    # our prometheus data representation
+    pdata = ""
+    for key, val in metrics.iteritems():
+        if key in skip:
+            continue
+        # add exporter name as a prefix for each key
+        key = '{}_{}'.format(exporter_name, key)
+        mhelp = "# HELP {}\n".format(key)
+        if isinstance(val, list):
+            mtype = "# TYPE {} histogram\n".format(key)
+            pdata += mhelp
+            pdata += mtype
+            for wdict in val:
+                entries = []
+                for kkk, vvv in wdict.iteritems():
+                    entries.append('{}="{}"'.format(kkk, vvv))
+                entry = "{%s}" % ','.join(entries)
+                pdata += "{}{} 1\n".format(key, entry)
+        elif isinstance(val, (str, tuple)):
+            continue
+        else:
+            pdata += mhelp
+            if isinstance(val, (int, long)):
+                mtype = "# TYPE {} counter\n".format(key)
+            if isinstance(val, float):
+                mtype = "# TYPE {} gauge\n".format(key)
+            pdata += mtype
+            pdata += "{} {}\n".format(key, val)
+    return pdata
+
+
+def flattenStats(cpdata):
+    "Flatten cherrypy stats to make them suitable for monitoring"
+    if isinstance(cpdata, str):
+        cpdata = json.loads(cpdata)
+    data = {}
+    for key, val in cpdata.iteritems():
+        if key.lower().find('cherrypy') != -1:
+            for kkk, vvv in val.iteritems():
+                nkey = 'cherrypy_app_%s' % kkk
+                nkey = nkey.lower().replace(" ", "_").replace("/", "_")
+                data[nkey] = vvv
+        if key.lower().find('cheroot') != -1:
+            for kkk, vvv in val.iteritems():
+                if kkk == 'Worker Threads':
+                    wdata = []
+                    for wkey, tval in vvv.iteritems():
+                        wkey = wkey.lower().replace(" ", "_").replace("/", "_").replace("-", "_")
+                        tval['thread_name'] = wkey
+                        nval = {}
+                        for tkey, tvvv in tval.iteritems():
+                            tkey = tkey.lower().replace(" ", "_").replace("/", "_")
+                            nval[tkey] = tvvv
+                        wdata.append(nval)
+                    data["cherrypy_server_worker_threads"] = wdata
+                else:
+                    nkey = 'cherrypy_http_server_%s' % kkk
+                    nkey = nkey.lower().replace(" ", "_").replace("/", "_")
+                    data[nkey] = vvv
+    return data
+
+
+class MetricsServer(object):
+    """
+    Example of metrics server which can server cherrypy metrics
```

Comment:
A typo here. Should be:
`     Example of metrics server which can serve cherrypy metrics`

Reply:
done